src/user/user.controller.ts

```diff
@@ -26,16 +29,31 @@ export class UserController {
     }
   }
 
+  @Get('profile')
+  @UseGuards(JwtAuthGuard)
+  async getProfile(@Req() req: Request & { user: any }) {
+    if (!req.user || !req.body.email) {
+      throw new BadRequestException('Invalid user data.');
+    }
+    return this.userService.findByEmail(req.body.email);
+  }
+  
+
+@Get(':id')
+findOne(@Param('id') id: string) {
+  const numericId = Number(id);
```

Comment:
You can avoid doing this casting manually using the built-in pipe [`ParseIntPipe`](https://docs.nestjs.com/pipes#built-in-pipes)

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/21/commits/f9f03caa56135c9e5e686147c7d439fe5d342767